BWEnv/src/controller.cc

```diff
@@ -782,17 +782,11 @@ void Controller::addUnit(BWAPI::Unit u, replayer::Frame& frame, BWAPI::PlayerInt
     + utype.airWeapon().damageBonus()
     * player->getUpgradeLevel(utype.airWeapon().upgradeType()))
     * utype.maxAirHits() * utype.airWeapon().damageFactor();
-  int max_health = utype.maxHitPoints();
```

Comment:
max_health wasn't used anywhere besides being passed to the unit constructor. BWAPI::UnitType::maxHitPoints already returns an int, so it can't be a casting issue. Now it just has maxHitPoints being passed directly to the constructor.

Calling getHitPoints() on minerals / geysers should return 1, so that shouldn't cause any issues.